server/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -1819,6 +1819,12 @@ private enum ElasticsearchExceptionHandle {
             org.elasticsearch.action.search.VersionMismatchException::new,
             161,
             Version.V_7_12_0
+        ),
+        HTTP_HEADERS_VALIDATION_EXCEPTION(
+            org.elasticsearch.http.HttpHeadersValidationException.class,
+            org.elasticsearch.http.HttpHeadersValidationException::new,
+            162,
+            Version.V_7_17_11
```

Comment:
I've pushed [140f323](https://github.com/elastic/elasticsearch/pull/96703/commits/140f323b85b2878b1e11edbc92fc114382ea4fc8) to make the `HttpHeadersValidationException` not extend `ElasticsearchException` such that it doesn't have to be registered for serialization. This is possible because `HttpHeadersValidationException`s are never serialized across nodes (they are exclusively used to "communicate" between the validating channel handler and the `RestController`).

(I turns out backporting new `ElasticsearchException` types to 7.17 is not easily done).